Taskfile.yml

```diff
@@ -2,28 +2,92 @@
 
 version: '3'
 
+env:
+  OPSLEVEL_GO_PKG: "github.com/opslevel/opslevel-go/v2024"
+  SRC_DIR: "{{.TASKFILE_DIR}}"
+
 vars:
-  # 'VERSION_DYNAMIC' needs to run in this scope before 'task build' begins
+  # 'VERSION_DYNAMIC' needs to run in this scope before 'task terraform-build' begins
   VERSION_DYNAMIC:
     sh: echo "$(date +%s).0.0"
   WORKSPACE_DIR: "workspace"
 
-includes:
-  go:
-    internal: true
-    taskfile: "./opslevel/Taskfile.yml"
```

Comment:
What is the motivation / benefit behind moving the contents of the opslevel (Go specific) Taskfile into this root level Taskfile?

Reply:
@davidbloss Overall I do not see the benefit in having 2 Taskfiles. I understand that the benefits of having two Taskfiles is that we get a separation of responsibilities where the Go Taskfile handles Go-specific things only.

However I do not think that that is enough of a benefit compared to the drawbacks.

1. It's harder to keep the Taskfile and Go-specific Taskfile in sync. If you delete or rename a task from file A it's not easy to remember to update the other Taskfile. I understand that you can use refactoring tools and plugins in an IDE to do this, but that adds overhead to something that should be simple.
2. It's harder to read the Taskfile's order of execution and to figure out what's doing what because you have to jump between multiple files. 
3. All of our other tools that are using Taskfiles are using a single file. This makes this repo's Taskfile the odd one out, and makes keeping it in sync with Taskfiles from other even repos harder.
4. Adding to this, it would be really annoying if each repo started having its own nested Taskfiles that are slightly different from each other. Taskfiles are not long and complex enough to warrant them not all being nearly identical.
5. I would prefer that tasks are differentiated by their name (e.g. terraform stuff is prefixed with `terraform-`) rather than what file they are contained in. It is much easier to understand what `task terraform-apply` does from its name compared to `task apply`.
6. The separation of responsibility isn't so clear. It made sense when we had Taskfile 1 that had all the basic commands (like `ci, setup, lint, fix, workspace`) and then Taskfile 2 for terraform commands and Taskfile 3 for Go commands. But now Taskfile 1 is doing basic commands AND terraform whereas Taskfile 2 is specific to only one thing. If we added another dependency that doesn't relate to Go/Terraform, should it be placed into its own separate Taskfile, or should it go into Taskfile 1? By placing it into its own Taskfile it would make our Taskfile hierarchy even more painful to maintain, which is why we removed the terraform Taskfile already. By placing it into the root Taskfile, the root Taskfile would seem crowded in comparison compared to the Go Taskfile since it's handling 3 things at once.

TL;DR I value uniformity of Taskfiles more than trying to separate them by responsibility.

Reply:
These Taskfiles are supposed to be a utility to make developing easier, not to become the an end in themselves.

On the drawbacks:

1. This surprises me, we're working with two files here and this sounds fairly hypothetical. Code reviews help to catch any rename/delete misses. If this truly stressful work I would be happy to take Taskfile related tickets in the future. Plugins and refactoring tools sound like overkill for this kind of work. I usually opt for a vertically split window. Also, take a look at Gitlab's web of [nested yml files](https://github.com/gitlabhq/gitlabhq/blob/master/.gitlab/ci/rails.gitlab-ci.yml) - it could be much worse.
2. IMO it's actually harder to scroll up and down searching through an extra 100 lines of code. Viewing two distant parts of a long file, or two parts of two different files can be solved by a split window.
3. True, the other repos have a single Taskfile, but all of our other tools are purely Go code bases. For this repo with Go and Terraform code, we're primarily interested in configuring/running Terraform operations and need to build the Go code as a necessary step. This repo is the odd one out by design and the Taskfile's match that. 
4. I didn't think our code bases were big enough for this concern. 🤷‍♂️ 
5. 👍 Certainly, whatever makes our work easier. I see you added aliases to those tasks.
6. I think this should have been a meeting or at least a discussion earlier on. I had no idea this was causing such a headache. The terraform Taskfile was buggy due to the working directory that we were running tasks in. Before, `task plan` just ran `terraform plan`. With `terraform plan` we have to be in the right directory, with `task plan` the idea was we could run `terraform plan` from any directory. Consolidating the terraform Taskfile into the root Taskfile was to get around all the directory changing and careful use of these [special variables](https://taskfile.dev/api/#special-variables).

I understand that we all have preferences but this is a pretty hefty change for essentially the same command line UX and feature set. Please lmk if there is anything else like this that's impeding your work.

Reply:
@davidbloss fair point regarding the nested yml files. I am going to revert the merge but keep the new aliases and tasks.

Reply:
@davidbloss I unmerged everything now there's two taskfiles again but I still require use of the special variables.

Reply:
resolving conversation - 1 taskfile is the way to go because controlling directories is very difficult AND there isn't really a reason to run a taskfile in the opslevel/ dir and not the root dir.